src/components/pages/variables.jsx

```diff
@@ -1,28 +1,55 @@
 import DocumentTitle from "react-document-title"
 import React, {PropTypes} from "react"
 import {Link} from "react-router"
-import {append, intersperse, lensPath, over, prepend, reduce, sortBy, toPairs} from "ramda"
+import {all, append, intersperse, isNil, lensPath, map, merge, over, pipe, prepend, reduce, reject, sortBy, toPairs, values} from "ramda"
 import TreeView from 'react-treeview'
 
 import * as AppPropTypes from "../../app-prop-types"
 import BreadCrumb from "../breadcrumb"
 import GitHubLink from "../github-link"
 import List from "../list"
+import SearchBox from "../search-box"
 
 
-function buildVariablesTree(variables) {
+function buildVariablesTree(variables, nameFilter, labelFilter) {
   const getVariablePath = variable => prepend('children', intersperse("children", variable.source_file_path.split("/")))
-  return reduce(
-    (memo, variable) => over(
-      lensPath(getVariablePath(variable)),
-      node => node
-        ? over(lensPath(["variables"]), append(variable), node)
-        : {variables: [variable]},
-      memo,
-    ),
-    {},
-    variables
+  const matchFilters = variable => (!nameFilter || variable.name.toLowerCase().includes(nameFilter.toLowerCase())) && (
+    !labelFilter || variable.label && variable.label.toLowerCase().includes(labelFilter.toLowerCase())
```

Comment:
I'm saying [accent-folding](http://alistapart.com/article/accent-folding-for-auto-complete) should be part of #23 :wink: